File: python/quick_sort.py
# 快速排序算法

def quick_sort(arr):
    if len(arr) < 2:
        return arr
    else:
        pivot = arr[0]
        less = [i for i in arr[1:] if i <= pivot]
        greater = [i for i in arr[1:] if i > pivot]
        return quick_sort(less) + [pivot] + quick_sort(greater)

if __name__ == "__main__":
    list = [2, 115, 8, 90]
    print(quick_sort(list))

File: java/QuickSort.java
/**
 * 快速排序算法  分而治之的策略
 */
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;

public class QuickSort{
    public static void main(String[] args){
        Integer[] arr = {2, 56, 1, 34, 12, 76, 45};
        List<Integer> list = Arrays.asList(arr);
        System.out.println(quickSort(list));
    }

    public static List<Integer> quickSort(List<Integer> list){
        if(list.size() < 2){
            return list;
        }else{
            int pivot = list.get(0);
            List<Integer> less = new ArrayList<Integer> ();
            List<Integer> greater = new ArrayList<Integer> ();
            for(int i = 1; i < list.size(); i++){
                if(list.get(i) <= pivot)
                    less.add(list.get(i));
                else
                    greater.add(list.get(i));
            }
            less = quickSort(less);
            less.add(pivot);
            less.addAll(quickSort(greater));
            return less;
        }
    }
}